lib.php

```diff
@@ -671,6 +702,18 @@ function local_augmented_is_excluded($userid, $courseid) {
     return $DB->record_exists_sql($sql, array($userid, $courseid, time()));
 }
 
+function local_augmented_is_suspended($user, $user_enrolments) {
+    $status = false;
+
+    foreach ($user_enrolments as $user_enrolment) {
+        if ($user_enrolment->userid === $user->id && (int)$user_enrolment->status === 1) {
```

Comment:
This if statement could be changed a bit for simplicity.

E.g. something like this:
if ($user_enrolment->userid === $user->id && $this->enrolment_is_active($user_enrolment))